Add channelResponseCallback to onJoinChannel action.

RamanQDO opened the pull request:
* we need to have a possibilities to track onChannelJoin responses (according to **hdo_assessor** backend architectures)
* also we need to have an opportunity to trigger some action in case of onChannelJoin failed (for example to show error)

According to existing implementation we only get default JOIN_FAILED response, which is very complex to differentiate from other  onChannelJoin events (we could have plenty of them in the same time)

BiruliaRaman committed:
Add channelResponseCallback to onJoinChannel action.

For now getPhoenixChannel method takes "channelResponseEvent" and "channelErrorResponseEvent" string params and dispatch corresponding actions as a callback.

BiruliaRaman committed:
fixup! Add channelResponseCallback to onJoinChannel action.

kuzaxak commented on src/middlewares/phoenix/actions/index.js:
```diff
@@ -201,8 +201,11 @@ export function connectToPhoenixChannelForEvents({
           })
         );
-        dispatch(endPhoenixChannelProgress({ channelTopic, loadingStatusKey: channelTopic }));
+        dispatch(
+          endPhoenixChannelProgress({ dispatch, channelTopic, loadingStatusKey: channelTopic })
+        );
```
Please do not use fixed for formatting commits. Formatting is a separated logic that should be added as a separate commit.

kuzaxak commented on src/middlewares/phoenix/actions/index.js:
```diff
@@ -201,8 +201,11 @@ export function connectToPhoenixChannelForEvents({
+          endPhoenixChannelProgress({ dispatch, channelTopic, loadingStatusKey: channelTopic })
+        );
         if (channelResponseEvent) {
+          console.log(channelResponseEvent, dispatch, response);
```
You can modify local copy to add additional logging, it isn't require pushing to dependency repo

Commit:
Do not exit function before events bind

If channel isn't created yet we will never bind an events to it.

Commit:
Rename data to conventional payload in dispatch

Regarding [guideline][1] we should use `payload` as a data key inside actions.

`Should always put their data into a payload field`

[1]: https://redux.js.org/style-guide/style-guide#write-actions-using-the-flux-standard-action-convention
